packages/lit-html/src/lib/lit-html.ts

```diff
@@ -655,6 +661,54 @@ export class NodePart {
     }
   }
 
+  private __commitIterable(value: Iterable<unknown>): void {
+    // For an Iterable, we create a new InstancePart per item, then set its
+    // value to the item. This is a little bit of overhead for every item in
+    // an Iterable, but it lets us recurse easily and efficiently update Arrays
+    // of TemplateResults that will be commonly returned from expressions like:
+    // array.map((i) => html`${i}`), by reusing existing TemplateInstances.
+
+    // If value is an array, then the previous render was of an
+    // iterable and value will contain the NodeParts from the previous
+    // render. If value is not an array, clear this part and make a new
+    // array for NodeParts.
+    if (!isArray(this.__value)) {
+      this.__value = [];
+      this.__clear();
+    }
+
+    // Lets us keep track of how many items we stamped so we can clear leftover
+    // items from a previous render
+    const itemParts = this.__value as NodePart[];
+    let partIndex = 0;
+    let itemPart: NodePart | undefined;
+
+    for (const item of value) {
+      if (partIndex === itemParts.length) {
+        // If no existing part, create a new one
+        itemParts.push(
+          (itemPart = new NodePart(
+            this.__insert(createMarker()),
+            this.__insert(createMarker()),
+            this.options
+          ))
+        );
+      } else {
+        // Reuse an existing part
+        itemPart = itemParts[partIndex];
+      }
+      itemPart.__setValue(item);
```

Comment:
Let's make sure to make an issue about resolving the visibility of the part API.

Reply:
https://github.com/Polymer/lit-html/issues/1261